ci/tools/build-utils.js

```diff
@@ -1,66 +1,124 @@
-﻿import os from 'node:os';
-import { promises as fs } from 'node:fs';
 import path from 'node:path';
-import process from 'node:process';
+import process from "node:process";
+import fs from 'node:fs/promises';
+
+import { mkdist } from 'mkdist';
 import uglify from 'uglify-js';
-import { calcIntegrity } from './integrity-utils.js';
 
-const
-	isCDN = process.argv.some(a => a === '--cdn'),
-	SRC = 'src',
-	DIST = 'dist',
-	CDN = 'cdn',
-	filesToCopy = ['object-observer.js', 'object-observer.d.ts'],
-	filesToMinify = ['object-observer.js'];
+import { calcIntegrity } from "./integrity-utils.js";
+import * as stdout from './stdout.js';
+
+const SRC_DIR = 'src';
+const DIST_DIR = 'dist';
+
+const buildCDN = process.argv.some(a => a === '--cdn')
 
-process.stdout.write(`\x1B[32mStarting the build...\x1B[0m${os.EOL}`);
-process.stdout.write(os.EOL);
+stdout.writeGreen('Starting the build...')
+stdout.writeNewline();
+stdout.writeNewline();
 
-await ensureCleanDir(DIST);
+await cleanDistDir()
 
-process.stdout.write(`\tcopying "${SRC}" to "${DIST}"...`);
-for (const fileToCopy of filesToCopy) {
-	await fs.copyFile(path.join(SRC, fileToCopy), path.join(DIST, fileToCopy));
+await buildCJSModule();
+await buildESModule();
+
+if (buildCDN) {
+	await buildCDNResources();
 }
-process.stdout.write(`\t\x1B[32mOK\x1B[0m${os.EOL}`);
-
-process.stdout.write('\tminifying...');
-const options = {
-	toplevel: true
-};
-for (const fileToMinify of filesToMinify) {
-	const fp = path.join(DIST, fileToMinify);
-	const mfp = path.join(DIST, fileToMinify.replace(/\.js$/, '.min.js'));
-	const fc = await fs.readFile(fp, { encoding: 'utf8' });
-	const mfc = uglify.minify(fc, options).code;
-	await fs.writeFile(mfp, mfc);
+
+async function cleanDistDir() {
+	stdout.write(`\tcleaning "dist"...`);
+
+	await fs.rm(DIST_DIR, { recursive: true, force: true });
+	await fs.mkdir(DIST_DIR);
+
+	stdout.writeGreen('OK');
+	stdout.writeNewline();
 }
-process.stdout.write(`\t\t\t\x1B[32mOK\x1B[0m${os.EOL}`);
 
-if (isCDN) {
-	await buildCDNDistro(DIST, CDN);
-	const sriMap = await calcIntegrity(path.join(DIST, CDN));
-	await fs.writeFile('sri.json', JSON.stringify(sriMap, null, '\t'), { encoding: 'utf-8' });
+async function buildCJSModule() {
+	stdout.write('\tbuilding CJS resources...');
+
+	const { writtenFiles } = await mkdist({
+		cleanDist: false,
+		srcDir: SRC_DIR,
+		distDir: path.join(DIST_DIR, 'cjs'),
+		ext: 'js',
+		format: 'cjs',
+		pattern: '**/*.js'
+	});
+
+	await minify(writtenFiles);
+
+	stdout.writeGreen('OK');
+	stdout.writeNewline();
 }
 
-process.stdout.write(os.EOL);
-process.stdout.write(`\x1B[32mDONE\x1B[0m${os.EOL}`);
+async function buildESModule() {
+	stdout.write('\tbuilding ESM resources...');
+
+	const { writtenFiles } = await mkdist({
+		cleanDist: false,
+		srcDir: SRC_DIR,
+		distDir: path.join(DIST_DIR),
+		ext: 'mjs',
```

Comment:
Probably this can stay `js` as well? The package has the type as module, so should be treated as ESM module by default.

Reply:
Yes, I haven't been sure about this so I left it as `mjs` but will change it to `js`.